apps/molgenis-components/src/components/forms/InputRefSelect.vue

```diff
@@ -0,0 +1,182 @@
+<template>
+  <FormGroup :id="id" :label="label" :description="description">
+    <InputGroup>
+      <template v-slot:append>
+        <button
+          v-if="value"
+          @click="$emit('input', null)"
+          class="btn btn-outline-primary"
+          type="button"
+        >
+          <i class="fas fa-fw fa-times"></i>
+        </button>
+      </template>
+      <input type="hidden" class="form-control" />
+      <select
+        class="form-control"
+        :id="id"
+        @change.stop.prevent="showSelect = true"
+        @click.stop.prevent="showSelect = true"
+      >
+        <option v-if="!showSelect"  :value="value">
+          {{ refLabel ? applyJsTemplate(value) : flattenObject(value) }}
+        </option>
+      </select>
+
+      <LayoutModal
+        v-if="showSelect"
+        :title="title"
+        @close="showSelect = false"
+      >
+        <template v-slot:body>
+          <TableSearch
+            :lookupTableName="tableName"
+            :filter="filter"
+            :graphqlURL="graphqlURL"
+            @select="select($event)"
+            @deselect="deselect(selectIdx)"
+          >
+            <template v-slot:rowheader="slotProps">
+              <ButtonAction @click="select(slotProps.rowkey)">
+                Select
+              </ButtonAction>
+            </template>
+          </TableSearch>
+        </template>
+        <template v-slot:footer>
+          <ButtonAlt @click="showSelect = false">Close</ButtonAlt>
+        </template>
+      </LayoutModal>
+    </InputGroup>
+  </FormGroup>
+</template>
+
+<script>
+import BaseInput from "./baseInputs/BaseInput.vue";
+import TableSearch from "../tables/TableSearch.vue";
+import LayoutModal from "../layout/LayoutModal.vue";
+import FormGroup from "./FormGroup.vue";
+import ButtonAlt from "./ButtonAlt.vue";
+import ButtonAction from "./ButtonAction.vue";
+import { flattenObject } from "../utils";
+
+export default {
+  name: "InputRefSelect",
+  extends: BaseInput,
+  data: function () {
+    return {
+      showSelect: false,
+    };
+  },
+  components: {
+    TableSearch,
+    LayoutModal,
+    FormGroup,
+    ButtonAction,
+    ButtonAlt,
+  },
+  props: {
+    tableName: String,
+    graphqlURL: {
+      default: "graphql",
+      type: String,
+    },
+    filter: Object,
+    refLabel: String,
+  },
+  computed: {
+    title() {
+      return "Select " + this.table;
+    },
+  },
+  methods: {
+    flattenObject(objectToFlatten) {
```

Comment:
Lots of components use this function. It has been moved to the utils file
/components/utils.js

Reply:
it used the flattenObject form utils